tests/process_test_set.py

```diff
@@ -276,6 +277,26 @@ def can_write_to_stdin_of_spawned_process_when_using_pty(shell):
         # Get the output twice since the pty echoes input
         assert_equal(b"hello\r\nhello\r\n", result.output)
 
+    @test
+    def exception_is_raised_if_cwd_does_not_exist(shell):
+        try:
+            shell.run(["echo", "run"], cwd="/some/silly/path")
+            assert_true(False)
+        except spur.RunProcessError as error:
+            # SshShell
+            assert_true(b'No such file or directory' in error.stderr_output)
+        except (IOError, OSError) as e:
+            # LocalShell
+            # FileNotFoundError on Py3.3+ inherits from IOError
+            # OSError is raised on Python 2
+            # errno.ENOENT: No such file or directory
+            assert_equal(e.errno, errno.ENOENT)
```

Comment:
There's already a Python exception for this kind of error, maybe it's better to use that instead of creating a new one?
Shouldn't we raise what Python would raise?
Problem is compatibility. It won't be the same on Python 2 and 3 (`FileNotFoundError`is only available in Python >= 3.3).